includes/html/dev-overview-data.inc.php

```diff
@@ -126,9 +126,15 @@
         <div class="col-sm-8"><span id="coordinates-text">' . $location_coords . '</span><div class="pull-right">';
 
     echo '<button type="button" id="toggle-map-button" class="btn btn-primary btn-xs" data-toggle="collapse" data-target="#toggle-map"><i class="fa fa-map" style="color:white" aria-hidden="true"></i> <span>View</span></button>';
-    if ($location_valid) {
-        echo ' <a id="map-it-button" href="https://maps.google.com/?q=' . $location->lat . ',' . $location->lng . '" target="_blank" class="btn btn-success btn-xs" role="button"><i class="fa fa-map-marker" style="color:white" aria-hidden="true"></i> Map</a>';
+
+   # Display Map button which will take you to Google or OSM depending on the setting map_provider in config.php
+    echo ' <a id="map-it-button" href="';
+    if ($location_valid && (Config::get('map_provider') != 'osm')) {
```

Comment:
if location isnt valid we end up with a broken link

Reply:
Good point, thanks Jellyfrog, I will check it and resubmit